### PR TITLE
TCP relay

### PR DESCRIPTION
**Pull request by kurnevsky**

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -42,7 +53,7 @@ fn bind_socket(addr: SocketAddr) -> UdpSocket {
 }
 
 /// Save DHT keys to a binary file.
-fn save_keys(keys_file: String, pk: PublicKey, sk: SecretKey) {
+fn save_keys(keys_file: &String, pk: PublicKey, sk: SecretKey) {
```
Comment (kpp) [inline src/main.rs]: BTW, it does not have to accept PK in args

File: src/main.rs [hunk 2]
```diff
@@ -104,30 +115,151 @@ fn run<F>(future: F, threads_config: ThreadsConfig)
...
+    };
+    let udp_onion = UdpOnion {
+        tx: udp_onion_tx,
+        rx: tcp_onion_rx,
+    };
+    (tcp_onion, udp_onion)
+}
+
+fn run_tcp(cli_config: &CliConfig, dht_sk: SecretKey, tcp_onion: TcpOnion) -> impl Future<Item = (), Error = Error> {
+    let tcp_server = TcpServer::new_with_onion(tcp_onion.tx);
```
Comment (kpp) [inline src/main.rs]: Maybe add setter for onion as it was done for dht - `set_tcp_onion_sink`?

File: src/main.rs (hunk 1 above)
Comment (kurnevsky) [reply src/main.rs]: It has to since this function saves it as well.

File: src/main.rs (hunk 2 above)
Comment (kurnevsky) [reply src/main.rs]: Yes, but this will require another pr in https://github.com/tox-rs/tox-rs project.

File: src/main.rs (hunk 1 above)
Comment (kpp) [reply src/main.rs]: You may get it from sk.

Comment (kurnevsky) [reply src/main.rs]: Not get but compute. It's computationally expensive operation so it's better to reuse keys where we can.

Comment (kpp) [reply src/main.rs]: > expensive operation

You do it only once. It does not affect on the speed of the server at all.

Comment (kurnevsky) [reply src/main.rs]: But why do it at all when I can just pass it as a parameter? As easy as that. This function even doesn't have too many arguments.

File: src/main.rs (hunk 2 above)
Comment (kpp) [reply src/main.rs]: Do it please.

Comment (kpp) [reply src/main.rs]: https://github.com/tox-rs/tox/pull/173 merged

File: .travis.yml [hunk 3]
```diff
@@ -1,6 +1,6 @@
 language: rust
 rust:
-  - 1.25.0
+  - 1.26.0
```
Comment (kpp) [inline .travis.yml]: Why?

Comment (kurnevsky) [reply .travis.yml]: As we discussed, 1.25 will require boxes here: https://github.com/tox-rs/tox-node/blob/master/src/main.rs#L190-L192
Besides that I used `impl trait` that is also was stabilized in 1.26.

Comment (kpp) [reply .travis.yml]: As we discussed, it should be noted in README

Comment (kurnevsky) [reply .travis.yml]: But we don't have mention of rust version in readme of `tox-node`. Shall I add it?

Comment (kpp) [reply .travis.yml]: Yes please

**Pull request merged**